design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,180 @@
+
+For any claim being mapped we will add the host annotation to the `mandatory_claims` variable and not the JWT claim name.
+
+
+## Design
+[//]: # "Add any diagrams, charts and explanations about the design aspect of the solution. Elaborate also about the expected user experience for the feature"
+
+The following class diagram, represent the flow of checking the mandatory claims and the claim mapping.
+
+### Class Diagrams
+[//]: # "Describe classes that are going to be added /changes and their immediate environment. Non-changed classes may be colored differently"
+
+![image-20210711122753500](token-schema-class-diagram.png)
+
+### Flow Explanation
+
+Upon validate_restrictions is called in JWTVendorConfiguration class these what would happen:
+
+1. The `valitate_restrictions` function will call `CreateConstraintsFromPolicy` that create constraints object checking the hosts annotations.
+   1. It will create instance of `NonEmptyConstraint` to check we have at least one host annotation.
+   2. It will call `LoadMAndatoryClaims` command class to load the list of mandatory claims for the `mandatory_claims` variable
+   3. It will create `RequiredContraint` object from it
+2. The `validate_restrictions` functon will call `CreateValidatorFromPolicy` to create validator with the claims mapping loaded
+   1. It will call the `LoadClaimsMapping` command class to get dictionary between claim to annotation.
+   2. It will create *ValidateRestrictionOneToOne* object and will return it
+3. `ValidateResourceRestrictions` will be called with the contraints and the validator from the previous steps. This run will check the claims in the token and annotations. If anything wrong like a missing claim in the host annotation or claim with invalid value and error with be thrown and the authentication request will be denied.
+
+## Backwards Compatibility
+[//]: # "How will the design of this solution impact backwards compatibility? Address how you are going to handle backwards compatibility, if necessary"
+
+Token schema is not mandatory therfore the JWT authentication will remain backwards compatible.
+
+## Affected Components
+
+[//]: # "List all components that will be affected by your solution"
+[//]: # "[Conjur Open Source/Enterprise, clients, integrations, etc.]"
+[//]: # "and elaborate on the impacts. This list should include all"
+[//]: # "downstream components that will need to be updated to consume"
+[//]: # "new releases as these changes are implemented"
+
+* JWT Authenticator
+
+
+
+## Delivery Plan
+
+| Mission                                          | Estimation |
+| ------------------------------------------------ | ---------- |
+| Support mandatory claims                         | 5 SP       |
+| Support claims mapping                           | 5 SP       |
+| Add status checks for token schema configuration | 2 SP       |
+| Add Cucumber tests for token schema              | 5 SP       |
```

Comment:
what about a security review by Sec Champion and the team? _Especially_ since we are dealing with authentication

Reply:
I would highly advice that the code review is conducting live with the group to evaluate the areas of vulnerability and the sec boundary

Reply:
What about the following sections:
Demo , Docs, Security

Reply:
There is nothing to write there so why to add empty sections?